packages/commands/getschema/src/index.ts

```diff
@@ -0,0 +1,201 @@
+import * as fs from 'fs'
+import * as os from 'os'
+import * as mkdirp from 'mkdirp'
+import { relative, dirname } from 'path'
+import { printSchema, GraphQLSchema, buildClientSchema, validateSchema } from 'graphql'
+import chalk from 'chalk'
+import { Arguments } from 'yargs'
+
+import { defineCommand } from '@graphql-cli/common';
+
+export default defineCommand(() => {
+  return {
+    command: 'get-schema',
+    builder(builder: any) {
+      return builder.options({
+        endpoint: {
+          alias: 'e',
+          describe: 'Endpoint name or URL',
+          type: 'string',
+        },
+        json: {
+          alias: 'j',
+          describe: 'Output as JSON',
+          type: 'boolean',
+        },
+        output: {
+          alias: 'o',
+          describe: 'Output file name',
+          type: 'string',
+        },
+        console: {
+          alias: 'c',
+          describe: 'Output to console',
+          default: false,
+        },
+        insecure: {
+          alias: 'i',
+          describe: 'Allow insecure (self-signed) certificates',
+          type: 'boolean',
+        },
+        header: {
+          describe:
+            'Header to use for downloading (with endpoint URL). Format: Header=Value',
+          type: 'string',
+        },
+      });
+    },
+    handler,
+  };
+});
+
+const handler = async (args: any) => {
+  if (args.endpoint) {
+    args.all = false
+  }
+
+  if (args.all && !args.project) {
+    args.project = args.endpoint = '*'
+  }
+
+  if (args.insecure) {
+    process.env.NODE_TLS_REJECT_UNAUTHORIZED = '0'
+  }
+
+  if (!args.watch) {
+    await updateWrapper(args)
+  }
+}
+
+async function updateWrapper(argv: any) {
+  try {
+    await update(argv)
+  } catch (err) {
+    console.error(err)
+  }
+}
+
+async function update(argv: Arguments) {
+  if (argv.endpoint) {
+    if (argv.output || argv.console) {
+      await downloadFromEndpointUrl(argv)
+      return
+    } else {
+      console.log('No output file specified!');
+    }
+  }
+}
+
+async function downloadFromEndpointUrl(argv: Arguments) {
+  const endpointHeaders = {}
+  if (argv.header) {
+    const headers = Array.isArray(argv.header) ? argv.header : [argv.header]
+    Object.assign(
+      endpointHeaders,
+      ...headers.map(h => ({ [h.split('=')[0]]: h.split('=')[1] })),
+    )
+  }
+
+  const endpoint = {
+    url: argv.endpoint,
+    headers: endpointHeaders,
+  }
+
+  await updateSingleProjectEndpoint(endpoint, argv)
+}
+
+async function updateSingleProjectEndpoint(
+  endpoint: { url: string, headers: string[] },
+  argv: Arguments,
+): Promise<void> {
+  console.info(`Downloading introspection from ${chalk.blue(endpoint.url)}`)
+  let newSchemaResult
+  try {
+    newSchemaResult = argv.json
+      // TODO figure out how to resolve it using graphql-tools
+      ? await endpoint.resolveIntrospection()
```

Comment:
Most of the code will stay here anyway.. This is the part that needs to be swapped.